docs/developersetup.md

```diff
@@ -0,0 +1,56 @@
+# Developer Setup
+---
+This page contains information about how to set up your local environment to develop for this project. 
+
+## MacOS Setup
+---
+
+1. [Install Homebrew](https://brew.sh)
+2. [Install Docker](https://docs.docker.com/get-docker/)
+3. [Install Git](https://git-scm.com/downloads)
+4. Install IDE
+
+## Git Setup
+---
+
+### Setting up your local repo
+
+1. Clone repo locally with:
+`git clone https://github.com/mellychat/webapp.git`
+
+2. Install recommended extensions
+
+3. Setup remote to track upstream with:
+`git remote add upstream https://github.com/mellychat/webapp.git`
+
+### Workflow
+
+1. Fetch any changes from the upstream/mainline branch and merges them into your local copy of mainline:
+
+  `git fetch upstream`
+  `git checkout mainline`
+  `git merge upstream/mainline`
+
```

Comment:
Add a tip:

```suggestion
TIP: ONLY when on the `mainline` branch, you can use `git pull upstream mainline` instead of the previous three commands.
```